src/redux/orderSlice.ts

```diff
@@ -0,0 +1,252 @@
+import { createSlice, createAsyncThunk } from '@reduxjs/toolkit';
+import { CurrencyFormat } from 'util/planHelper';
+import { createOrder, getOrder, updateOrder } from '../api';
+import { MESSAGE_TYPE_FAIL, MESSAGE_TYPE_SUCCESS } from '../components/Input';
+import { RootState } from './rootReducer';
+
+type Order = {
+  billingAddress: unknown;
+  country: string;
+  couponId: unknown;
+  currency: CurrencyFormat;
+  customer: { locale: string; email: string };
+  customerId: number;
+  discount: { applied: boolean; type: string; periods: number };
```

Comment:
I think that we could add: type: 'coupon' | 'trial' (not sure if that's all options)